src/pybind/mgr/diskprediction/agent/predict/prediction.py

```diff
@@ -0,0 +1,308 @@
+                            break
+
+                tmp = {}
+                if wwpn:
+                    tmp['disk_domain_id'] = wwpn
+                    tmp['disk_wwn'] = wwpn
+                    if serial_number:
+                        tmp['serial_number'] = serial_number
+                    else:
+                        tmp['serial_number'] = wwpn
+                elif serial_number:
+                    tmp['disk_domain_id'] = serial_number
+                    tmp['serial_number'] = serial_number
+                    if wwpn:
+                        tmp['disk_wwn'] = wwpn
+                    else:
+                        tmp['disk_wwn'] = serial_number
+                else:
+                    tmp['disk_domain_id'] = dev_name
+                    tmp['disk_wwn'] = dev_name
+                    tmp['serial_number'] = dev_name
+
+                if s_val.get('smart_status', {}).get("passed"):
+                    tmp['smart_health_status'] = 'PASSED'
+                else:
+                    tmp['smart_health_status'] = 'FAILED'
+
+                tmp['sata_version'] = s_val.get('sata_version', {}).get('string', '')
+                tmp['sector_size'] = str(s_val.get('logical_block_size', ''))
+                try:
+                    if isinstance(s_val.get('user_capacity'), dict):
+                        user_capacity = \
+                            s_val['user_capacity'].get('bytes', {}).get('n', 0)
+                    else:
+                        user_capacity = s_val.get('user_capacity', 0)
+                except ValueError:
+                    user_capacity = 0
+                disk_info = {
+                    'diskName': dev_name,
+                    'diskType': str(disk_type),
+                    'diskStatus': '1',
+                    'diskWWN': tmp['disk_wwn'],
+                    'dpDiskId': tmp['disk_domain_id'],
+                    'serialNumber': tmp['serial_number'],
+                    'vendor': vendor,
+                    'sataVersion': tmp['sata_version'],
+                    'smartHealthStatus': tmp['smart_health_status'],
+                    'sectorSize': tmp['sector_size'],
+                    'size': str(user_capacity),
+                    'prediction': self._parse_prediction_data(host_domain_id, tmp['disk_domain_id'])
+                }
+                info_list.append(disk_info)
+                # Update osd life-expectancy
+                dev_id = ''
+                predicted = None
+                life_expectancy_day_min = None
+                life_expectancy_day_max = None
+                devs_info = obj_api.get_osd_device_id(osd_id)
+                if disk_info.get('prediction', {}).get('predicted'):
+                    predicted = int(disk_info['prediction']['predicted'])
+                if disk_info.get('prediction', {}).get('near_failure'):
+                    if disk_info['prediction']['near_failure'] == 'Good':

```

Comment:
Lower case this check if you do not have this value under your control and comes from external resource. smartmontools changed the writing for some values from one version to another for example.